src/rust/wpcomics/template/src/template.rs

```diff
@@ -0,0 +1,155 @@
+    Listing, Manga, MangaPageResult, Page, MangaStatus, MangaContentRating, MangaViewer, Chapter, DeepLink
+};
+
+use crate::helper::{i32_to_string, append_protocol, https_upgrade, extract_i32_from_string};
+
+pub fn get_manga_list(search_url: String, next_page_selector: String, title_transformer: fn(String) -> String) -> Result<MangaPageResult> {
+    let mut mangas: Vec<Manga> = Vec::new();
+    let mut has_next_page = next_page_selector.len() > 0;
+    let html = Request::new(&search_url, HttpMethod::Get).html();
+    for item in html.select("div.items > div.row > div.item > figure.clearfix").array() {
+        let item_node = item.as_node();
+        let title = item_node.select("figcaption > h3 > a").first().text().read();
+        let id = https_upgrade(item_node.select("div.image > a").first().attr("href").read());
+        let cover = append_protocol(item_node.select("div.image > a > img").first().attr("data-original").read());
+        mangas.push(Manga {
+            id,
+            cover,
+            title: title_transformer(title),
+            author: String::new(),
+            artist: String::new(),
+            description: String::new(),
+            url: String::new(),
+            categories: Vec::new(),
+            status: MangaStatus::Unknown,
+            nsfw: MangaContentRating::Safe,
+            viewer: MangaViewer::Default
+        });
+    }
+    if next_page_selector.len() > 0 {
+        has_next_page = html.select(&next_page_selector).array().len() > 0;
+    }
+    Ok(MangaPageResult {
+        manga: mangas,
+        has_more: has_next_page,
+    })
+}
+
+pub fn get_manga_listing(base_url: String, listing: Listing, next_page_selector: String, listing_mapping: fn(String) -> String, title_transformer: fn(String) -> String, page: i32) -> Result<MangaPageResult> {
+    let mut url = String::new();
+    url.push_str(&base_url);
+    if listing.name != String::from("All") {
+        url.push_str("/");
+        url.push_str(&listing_mapping(listing.name));
+        url.push_str("?page=");
+        url.push_str(&i32_to_string(page));
+    } else {
+        url.push_str("/?page=");
+        url.push_str(&i32_to_string(page));
+    }
+    get_manga_list(url, next_page_selector, title_transformer)
+}
+
+pub fn get_manga_details(id: String, status_from_string: fn(String) -> MangaStatus, title_transformer: fn(String) -> String) -> Result<Manga> {
+    let details = Request::new(id.clone().as_str(), HttpMethod::Get).html();
+    let title = details.select("h1.title-detail").text().read();
+    let cover = append_protocol(details.select("div.col-image > img").attr("src").read());
+    let author = details.select("ul.list-info > li.author > p.col-xs-8").text().read();
+    let description = details.select("div.detail-content > p").text().read();
+    let mut categories = Vec::new();
+    let mut nsfw = MangaContentRating::Safe;
+    let mut viewer = MangaViewer::Default;
```

Comment:
Should use RTL by default.

```suggestion
    let mut viewer = MangaViewer::Rtl;
```